db_test.go

```diff
@@ -315,7 +276,12 @@ func (s *SurrealDBTestSuite) TestFetch() {
 		s.NoError(err)
 		s.NotEmpty(res)
 
-		s.validateFriends(res, userSlice[0].Friends[0])
+		userSlice, err := marshal.SmartUnmarshal[testUserWithFriend[testUserWithFriend[interface{}]]](res, err)
```

Comment:
So this actually isnt deserialising the entire response into nested structs, only pointing to nested structs and deserialising there?

It's really cool that this works. I am wondering if it will be confusing though. From a user perspective, people just want to get their data with a single command. This requires picking each nested doc, and getting the response to populate the nested values.

It's certainly worth having the fetch example.

What do you think we should do? Do you think it's possible to reach the above simplicity without pointing at nested values explicitly?

Also noting, this will likely change with Row and Column

Reply:
I agree but I'm not sure how should I implement it according to the Rows and Columns.
is `Marshal.UnmarshalRaw` ok?

Reply:
Yeah, fair point about being able to fetch. Will merge, as it gives us context about how this is going to be currently handled. cc @timpratim about this way of handling fetch going into API